RealmSwift/SwiftUI.swift

```diff
@@ -645,6 +645,285 @@ extension EnvironmentValues {
     }
 }
 
+@available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *)
+private class ObservableAsyncOpenStorage: ObservableObject {
+    var cancellables = [AnyCancellable]()
+    var realm: Realm?
+
+    @Published var asyncOpenState: AsyncOpenState = .notOpen {
+        willSet {
+            objectWillChange.send()
+        }
+    }
+
+    func asyncOpen(configuration: Realm.Configuration) -> RealmPublishers.AsyncOpenPublisher {
+        return Realm.asyncOpen(configuration: configuration)
+            .onProgressNotification { asyncProgress in
+                let progress = Progress(totalUnitCount: 1)
+                if asyncProgress.isTransferComplete {
+                    progress.completedUnitCount = Int64(1)
+                    self.asyncOpenState = .progress(progress)
+                } else {
+                    progress.completedUnitCount = Int64(asyncProgress.fractionTransferred)
+                    self.asyncOpenState = .progress(progress)
+                }
+            }
+    }
+}
+
+/**
+`AsyncOpenState`is an enum representing different states from `AsyncOpen` and `AutoOpen` process
+*/
+public enum AsyncOpenState {
+    /// Initial state
+    case notOpen
+    /// Returns a realm when the async open process is done or in case there is no internet connection in AutoOpen
```

Comment:
An enum case doesn't do things or return values, and the docs should explain what it means to be in that state. Some rough attempts at that:

open: "The Realm has been opened and is ready for use. For AsyncOpen this means that the Realm has been fully downloaded, but for AutoOpen the existing local file may have been used if the device is offline."
progress: "The Realm is currently being downloaded from the server."
error: "Opening the Realm failed."